src/components/shared/SearchableSelect.tsx

```diff
@@ -0,0 +1,69 @@
+import { Select } from "@radix-ui/themes";
+import { useEffect, useState} from "react";
+import * as React from 'react';
+
+interface SearchableSelectProps {
+  selectPlaceholder: string;
+  items: { label: string; value: string }[];
+  onSelect: (selectedValue: string) => void;
+  ariaLabelForTrigger: string;
+  ariaLabelForContent: string;
+}
+
+/* If the language of the browser appears in the picker and at the same time
+ * editing is forbidden as long as the language was not selected, the user will be confused
+ * in case the preselected language matches their intended language and they don not change
+ * the picker value. This is why there is a placeholder here instead of a default value.
+ * */
+
+export default function SearchableSelect({
+  selectPlaceholder,
+  items,
+  onSelect,
+  ariaLabelForTrigger,
+  ariaLabelForContent
+}: SearchableSelectProps) {
+  const [open, setOpen] = useState(false);
+  const [value, setValue] = useState("");
+
+  useEffect(() => {
+    if (value) {
+      onSelect(value);
+    }
+  }, [value, onSelect]);
+
+
+  return (
+    <>
```

Comment:
suggestion (non-blocking): From what I see, this doesn't have to be wrapped in a `<></>` fragment